### PR TITLE
 omnisharp-roslyn: init at 1.32.8

### PR DESCRIPTION
**Pull request by tesq0**
###### Motivation for this change
Intellisense for c# 

###### Things done

- [x ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/omnisharp-roslyn/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+, mono5
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+
+  name = "omnisharp-roslyn-${version}";
+  version = "1.32.8";
+
+  outputs = [ "out" ];
```
Comment (joachifm) [inline pkgs/development/tools/omnisharp-roslyn/default.nix]: This is likely redundant/implied when there are no other outputs listed.

File: pkgs/development/tools/omnisharp-roslyn/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  src = fetchurl {
+    url = "https://github.com/OmniSharp/omnisharp-roslyn/releases/download/v${version}/omnisharp-mono.tar.gz";
+    sha256 = "0k2a4awmzb7ppll2skyzaa94n3hxqm35ffibl0sygldk3symzwgp";
+  };
+
+  buildInputs = [ makeWrapper ];
```
Comment (joachifm) [inline pkgs/development/tools/omnisharp-roslyn/default.nix]: Consider moving this to `nativeBuildInputs`, which is more appropriate for build-time only inputs.

**Review by joachifm — COMMENTED** [review]
Minor nits

**Comment by sib-null** [comment]
Would this resolve #44999 when specifying to use the Nix package installation?

File: pkgs/development/tools/omnisharp-roslyn/default.nix (hunk 1 above)
Comment (tesq0) [reply pkgs/development/tools/omnisharp-roslyn/default.nix]: Indeed, it was unneeded.

File: pkgs/development/tools/omnisharp-roslyn/default.nix (hunk 2 above)
Comment (tesq0) [reply pkgs/development/tools/omnisharp-roslyn/default.nix]: yep

**Comment by tesq0** [comment]
@sib-null Yes, it would. 
You would need to add this to your settings.json in vscode:
`"omnisharp.path": "full/path/to/omnisharp"` (like "/home/someuser/.nix-profile/bin/omnisharp")

**Comment by sib-null** [comment]
Can this be merged?

**Comment by mattfredwill** [comment]
Using this with 

`"omnisharp.path": "/run/current-system/sw/bin/omnisharp"`

In vscode is working fine, seems like it could be merged?

**Comment by sib-null** [comment]
@mattfredwill , I'm getting the following error when opening a C# project:

`Cannot open assembly '/home/bill/.nix-profile/bin/omnisharp': File does not contain a valid CIL image.`

Any ideas?

**Pull request merged**